python/GafferSceneUI/SceneViewUI.py

```diff
@@ -244,8 +244,51 @@ def __menuDefinition( self ) :
 			}
 		)
 
+		m.append( "/__visualiserDivider__", { "divider" : True } )
+
+		visScaleIsOther = True
+		visScalePlug = self.getPlug()["visualiserOrnamentScale"]
+		for scale in ( 1, 10, 100 ) :
+			isSelected = visScalePlug.getValue() == scale
+			if isSelected :
+				visScaleIsOther = False
+			m.append(
+				"/Visualiser Scale/%d" % scale,
+				{
+					"command" : functools.partial( lambda s, _ : visScalePlug.setValue( s ), scale ),
+					"checkBox" : isSelected
+				}
+			)
+
+		m.append( "/Visualiser Scale/__divider__", { "divider" : True } )
+		m.append(
+			"/Visualiser Scale/Other...",
+			{
+				"command" : functools.partial( self.__pickerOther, float, visScalePlug, "Visualiser Scale" ),
+				"checkBox" : visScaleIsOther
+			}
+		)
+
 		return m
 
+	@staticmethod
+	def __pickerOther( type_, plug, title, *unused ) :
+
+		value = GafferUI.TextInputDialogue(
```

Comment:
This seems a little primitive. It'd be nice to be constrained to entering numbers, and to have the virtual sliders and cursor niceties of a numericwidget. It'd also be really nice to be able to see the scale change in the viewer as you edit, to aid in finding the right number. Seems like it might not be all that much more code to open a custom dialogue with a NumericPlugValueWidget in it?

Reply:
Good idea - https://github.com/GafferHQ/gaffer/pull/3519/commits/ead7ad7280100b8e537741e2cd4878e797727935